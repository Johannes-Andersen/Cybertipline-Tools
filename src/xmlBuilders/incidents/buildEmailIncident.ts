import { XMLBuilder, XMLParser } from 'fast-xml-parser';
import type { EmailIncident } from '../../types';
import { buildEmail } from '../common/buildEmail';

const parser = new XMLParser({
  attributeNamePrefix: '@_',
  ignoreAttributes: false,
  allowBooleanAttributes: true,
});

const builder = new XMLBuilder({
  attributeNamePrefix: '@_',
  ignoreAttributes: false,
  suppressBooleanAttributes: false,
});

export const buildEmailIncident = (
  emailIncident: EmailIncident,
  keyName = 'emailIncident',
): string =>
  builder.build({
    [keyName]: {
      emailAddress: emailIncident.emailAddress?.map(
        (e) => parser.parse(buildEmail(e)).email,
      ),
      content: emailIncident.content,
      additionalInfo: emailIncident.additionalInfo,
    },
  });
