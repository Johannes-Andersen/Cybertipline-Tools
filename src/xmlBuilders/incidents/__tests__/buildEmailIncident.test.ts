import { describe, expect, it } from 'vitest';
import type { EmailIncident } from '../../../types';
import { buildEmailIncident } from '../buildEmailIncident';

describe('buildEmailIncident', () => {
  it('should build a email incident', () => {
    const emailIncident: EmailIncident = {
      emailAddress: [
        {
          email: 'test@test.com',
          verified: true,
        },
        {
          email: 'test2@test.com',
          verified: false,
        },
      ],
      content: 'content',
      additionalInfo: 'additional info',
    };

    const result = buildEmailIncident(emailIncident);

    expect(result).toMatchSnapshot();
  });
});
