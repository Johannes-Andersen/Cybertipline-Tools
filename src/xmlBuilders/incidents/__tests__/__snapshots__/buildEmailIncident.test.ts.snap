// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildEmailIncident > should build a email incident 1`] = `"<emailIncident><emailAddress verified="true">test@test.com</emailAddress><emailAddress verified="false">test2@test.com</emailAddress><content>content</content><additionalInfo>additional info</additionalInfo></emailIncident>"`;
